### PR TITLE
Analytics Implementation

### PR DESCRIPTION
**Pull request by nikhilAgarwalM**
Event Logging
- Added network, analytics, logger modules
- Implemented Analytics default, debug, cached providers
- Implemented ImmediateSyncStrategy.kt in Cached Providers 
- Added Analytics SDK configuration
- Initialised analytics SDK in MediaNetAdSDK.kt with dummy values
- Firing dummy events from Ad class

NOTE - TimedSyncStrategy.kt is not fully implemented yet

### CODE REVIEW

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt [hunk 1]
```diff
@@ -102,5 +107,25 @@ object MediaNetAdSDK {
...
+
+        val configuration = AnalyticsSDK.Configuration.Builder()
+            .setDebugMode(false)
+            .setAnalyticsUrl("https://logstash-gcpi.net") //TODO get this in config call
+            .enableEventCaching(true)
+            .enableEventSampling(true, samplingMap)
+            .build()
 
+        AnalyticsSDK.init(applicationContext, configuration)
+        AnalyticsSDK.pushEvent(Event(name = "Config_call_success", type = LoggingEvents.PROJECT.type))
```
Comment (dk-pathak) [inline MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt:123]: Please use constants file `Config_call_success`

File: analytics/build.gradle [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        sourceCompatibility JavaVersion.VERSION_1_8
+        targetCompatibility JavaVersion.VERSION_1_8
+    }
+    kotlinOptions {
+        jvmTarget = '1.8'
+    }
+}
+
+dependencies {
+    implementation "androidx.core:core-ktx:1.9.0"
```
Comment (dk-pathak) [inline analytics/build.gradle:29]: Use gradle constants for all the versions

File: analytics/src/main/java/com/app/analytics/AnalyticsSDK.kt [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+    private var customProviders: List<AnalyticsProvider> = emptyList()
+
+    fun init(context: Context, configuration: Configuration, providers: List<AnalyticsProvider> = mutableListOf()) {
+        if (isInitialised.not()) {
+            config = configuration
+            customProviders = providers
+            addProviders(context, providers)
+            isInitialised = true
+            sendPendingEventIfAny()
+            CustomLogger.debug(TAG, "Analytics SDK initialised")
```
Comment (dk-pathak) [inline analytics/src/main/java/com/app/analytics/AnalyticsSDK.kt:27]: Please ensure there should be only one logger class. Prebid also come with its own Logger methods.

**Review by dk-pathak — COMMENTED** [review]
SOme minor improvements mentioned in the comments. Rest Look good.

File: MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt (hunk 1 above)
Comment (nikhilAgarwalM) [reply MediaNetAdSDK/src/main/java/com/medianet/android/adsdk/MediaNetAdSDK.kt]: This will be removed once we get event name and keys

**Pull request merged**